geoproxy/main.go

```diff
@@ -95,42 +113,117 @@ func parseRequestBody(request *http.Request) requestPayloadStruct {
+}
 
-	default_condtion_url := os.Getenv("DEFAULT_CONDITION_URL")
+// Log the submit typeform payload and redirect url
+func logSubmitRequestPayload(requestionPayload submitRequestPayloadStruct, proxyUrl string) {
+	log.Printf("coordinate: %s, proxy_url: %s\n", requestionPayload.LatLng, proxyUrl)
+}
 
-	if err1 != nil || err2 != nil {
-		log.Printf("Error: Invalid lat-long pair.")
-		return default_condtion_url
+// Given string in form [0.0, 0.0], create and return coord struct
+func parseCoord(coord string) Coord {
+	splitCoord := strings.Split(coord, ",")
+	// We should have someething like {"[0.0", "0.0]"}
+	if len(splitCoord) != 2 {
+		// Returning error coord of 360, 360 for now
+		log.Printf("Error: Coordinate passed into json request should be of form [0.0, 0.0].")
+		return Coord{360, 360}
+	} else {
+		coord0 := []rune(splitCoord[0])
+		coord1 := []rune(splitCoord[1])
+
+		lat, errLat := strconv.ParseFloat(string(coord0[1:len(splitCoord[0])]), 64)
+		lng, errLng := strconv.ParseFloat(string(coord1[0:len(splitCoord[1])-1]), 64)
+
+		if errLat != nil || errLng != nil {
+			log.Printf("Error: Invalid lat-long coordinate.")
+			return Coord{360, 360}
+		}
+		return Coord{lat, lng}
 	}
+}
+
+func rangesOverlap(start1 float64, end1 float64, start2 float64, end2 float64) bool {
+	return math.Min(end1, end2) >= math.Max(start1, start2)
+}
 
-	// Lat-long mapping logic to get the env
-	for l1 := range sorted_lat {
-		if lat < float32(l1) {
-			var sorted_lng []int
-			for l2 := range data[l1] {
-			    sorted_lng = append(sorted_lng, l2)
+// Get the url(s) for given coordinates of view request
+func getViewProxyUrl(rawCoord1 string, rawCoord2 string) map[string]bool {
+	// Acts as a set of urls
+	urls := make(map[string]bool)
+
+	// Parse each coord
+	topLeft := parseCoord(rawCoord1)
+	bottomRight := parseCoord(rawCoord2)
+
+	// **Commented out to test with only one server associated with midpoint for now**
+	// *******************************************************************************
+	// // Add all urls of zones that touch the box of the coords
+	// for latRange := range data {
+	// 	if rangesOverlap(topLeft.Lat, bottomRight.Lat, latRange.Low, latRange.High) {
+	// 		for lngRange := range data[latRange] {
+	// 			if rangesOverlap(topLeft.Lng, bottomRight.Lng, lngRange.Low, lngRange.High) {
+	// 				// Check if we have already added url
+	// 				urlString := os.Getenv(data[latRange][lngRange])
+	// 				// url, exists := urls[urlString]
+	// 				if !urls[urlString] {
+	// 					urls[urlString] = true
+	// 				}
+	// 			}
+	// 		}
+	// 	}
+	// }
+	// *******************************************************************************
+
+	coord := Coord{(topLeft.Lat + bottomRight.Lat) / 2, (topLeft.Lng + bottomRight.Lng) / 2}
+
+	for latRange := range data {
+		if coord.Lat >= latRange.Low && coord.Lat < latRange.High {
+			for lngRange := range data[latRange] {
+				if coord.Lng >= lngRange.Low && coord.Lng < lngRange.High {
```

Comment:
should these comparisons both be inclusive? currently the lower end isn't inclusive and the higher range is

Reply:
The ranges are inclusive on low and exclusive on high like you said, since a request on the edge of a range (with lat or lng equal to high) needs to go to a definitive server. For example coord with latitude 0 with latitude coordRanges (-90, 0) and (0, 90) in the map needs to fit only in one range--the first. If the ranges were low and high inclusive, the request could go to either coordRange's server and then we lose track of which server that request would be in.